### PR TITLE
Bug 1801415: Migrate extensions/v1beta1 -> networking.k8s.io/v1beta1

### PR DESCRIPTION
**Pull request by ironcladlou**
https://bugzilla.redhat.com/show_bug.cgi?id=1801415

### CODE REVIEW

**Comment by Miciah** [comment]
Do we need to convert objects using the old API, as https://github.com/kubernetes/ingress-nginx/pull/4127/commits/84102eec2ba270f624c57023aab59aab4471178e#diff-ece58b3bebec02a6d3173c8d8099d298R904-R939 does?

**Comment by ironcladlou** [comment]
cc @deads2k @sttts @mfojtik can you help us understand what else we might need to do here (https://github.com/openshift/openshift-controller-manager/pull/83#issuecomment-606658312)?

File: pkg/route/ingress/ingress.go [hunk 1]
```diff
@@ -514,7 +514,7 @@ func newRouteForIngress(
 			Labels:       ingress.Labels,
 			Annotations:  ingress.Annotations,
 			OwnerReferences: []metav1.OwnerReference{
-				{APIVersion: "extensions/v1beta1", Kind: "Ingress", Controller: &t, Name: ingress.Name, UID: ingress.UID},
+				{APIVersion: "networking.k8s.io/v1beta1", Kind: "Ingress", Controller: &t, Name: ingress.Name, UID: ingress.UID},
```
Comment (sttts) [inline pkg/route/ingress/ingress.go:517]: what about existing objects with this owner ref?

**Comment by deads2k** [comment]
> Do we need to convert objects using the old API, as [kubernetes/ingress-nginx@84102ee#diff-ece58b3bebec02a6d3173c8d8099d298R904-R939](https://github.com/kubernetes/ingress-nginx/commit/84102eec2ba270f624c57023aab59aab4471178e#diff-ece58b3bebec02a6d3173c8d8099d298R904-R939) does?

In repos that you own, yes you do.  `kubectl convert` may be able to do it for you.

File: pkg/route/ingress/ingress.go (hunk 1 above)
Comment (deads2k) [reply pkg/route/ingress/ingress.go:517]: > what about existing objects with this owner ref?

They will still be fine until the old API is gone entirely, so you just have to scan for and update them.

**Comment by ironcladlou** [comment]
@deads2k 
>>Do we need to convert objects using the old API, as kubernetes/ingress-nginx@84102ee#diff-ece58b3bebec02a6d3173c8d8099d298R904-R939 does?

>In repos that you own, yes you do. kubectl convert may be able to do it for you.

I'm not sure I asked my question very clearly. When the extensions API group finally "goes away", what exactly happens to the existing persisted resources in etcd? Must this controller (or some operator) be actively migrating the extensions API resources to the new API group to ensure that when the old API is gone forever there's nothing persisted in etcd with the old group?

**Comment by ironcladlou** [comment]
@deads2k @sttts thoughts on https://github.com/openshift/openshift-controller-manager/pull/83#issuecomment-612913612?

**Comment by sttts** [comment]
> I'm not sure I asked my question very clearly. When the extensions API group finally "goes away", what exactly happens to the existing persisted resources in etcd? Must this controller (or some operator) be actively migrating the extensions API resources to the new API group to ensure that when the old API is gone forever there's nothing persisted in etcd with the old group?

Nothing to do for operators or other consumers about the persisted api version. The persisted objects will be migrated through kube-storage-version-migrator (which we have an operator for now due to etcd encryption). This must be done before kube-apiserver stops being able to read the old api group. I trust @soltysh to remind us when this must be done.

**Comment by ironcladlou** [comment]
Looks like this is ready to go then as far as I can tell. cc @openshift/sig-network-edge

File: pkg/route/ingress/ingress.go [hunk 2]
```diff
@@ -8,8 +8,8 @@ import (
 
 	"k8s.io/klog"
 
-	"k8s.io/api/core/v1"
-	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	v1 "k8s.io/api/core/v1"
```
Comment (danehans) [inline pkg/route/ingress/ingress.go:11]: Either drop `v1` or rename to `corev1`.

Comment (ironcladlou) [reply pkg/route/ingress/ingress.go]: Fixed

**Comment by danehans** [comment]
`imagestreams.image.openshift.io "pipeline" is forbidden: User "system:serviceaccount:ci:ci-operator" cannot get resource "imagestreams" in API group "image.openshift.io" in the namespace "ci-op-pb6fb9h4"`

CI flake?

/retest

**Pull request merged**